core/drivers/fabric-driver/server/events.ts

```diff
@@ -36,15 +36,16 @@ async function subscribeEventHelper(
         client.sendDriverSubscriptionStatus(ack_send_error, relayCallback);
     } else {
         const ack_send = new ack_pb.Ack();
-        if (newRequestId == requestId) {
-            // event being subscribed for the first time
-
-            // the event listener logic follows here
-            await registerListenerForEventSubscription(call_request, network_name);
-            // set the following ack_send message below, only if the event listener logic is successful; else set to an error message
-
-            ack_send.setMessage('Event subscription is successful!');
-            ack_send.setStatus(ack_pb.Ack.STATUS.OK);
+        if (newRequestId == requestId) {    // event being subscribed for the first time
+            // Start an appropriate type of event listener for this event subscription if one is not already active
+            const [listenerHandle, error] = await handlePromise(registerListenerForEventSubscription(call_request, network_name));
+            if (error) {
+                ack_send.setMessage('Event subscription succeeded but listener registration failed');
```

Comment:
Here can we also delete the subscription from database? and just say event subscription failed?
Otherwise what do user needs to do in case of this error? they can't subscribe again as db entry is there, so is it something that the driver would automatically try to register listener from db later?

Reply:
`Event subscription error: listener registration failed`
Can you update that message at line 43?